src/app/services/videoprocessing/video_processor.py

```diff
@@ -175,41 +183,59 @@ def run(self, cameraConfigPath, faceDetectionMethod):
+
+                        # Queue dewarping for each virtual camera
+                        for vc in self.virtualCameraManager.getVirtualCameras():
+                            vcBuffer = np.empty((vcOutputHeight, vcOutputWidth, channels), dtype=np.uint8)
+
+                            # Generate dewarping params for vc
+                            vcDewarpingParameters = self.__getVirtualCameraDewarpingParameters(vc, fisheyeCenter, cameraConfig)
+    
+                            # Virtual camera always need the most recent fisheye image
+                            dewarper.queueDewarping(currentFisheyeBufferId, vcBufferId, vcDewarpingParameters, vcBuffer)
+                            vcBuffers.append(vcBuffer)
+
+                        # Execute all queued dewarping for face detection and virtual cameras
+                        bufferId = 0
+                        while bufferId != NoQueuedDewarping:
+                            bufferId = dewarper.dewarpNextImage()
+                            
+                            if bufferId == fdBufferId:
+                                buffer, dewarpIndex = fdBufferQueue.popleft()
+                                fdBuffers[dewarpIndex] = buffer
+                                self.imageQueue.put_nowait((buffer, dewarpIndex))
                         
-                        if bufferId == fdBufferId:
-                            buffer, dewarpIndex = fdBufferQueue.popleft()
-                            fdBuffers[dewarpIndex] = buffer
-                            self.imageQueue.put_nowait((buffer, dewarpIndex))
-                    
-                    # Dislay 360 dewarping as virtual cameras for debugging
-                    if show360DegAsVcs:
-                        vcBuffers.extend(fdBuffers)
-
-                    # Send the virtual camera images
-                    self.signalVirtualCameras.emit(vcBuffers, self.virtualCameraManager.getVirtualCameras())
+                        # Dislay 360 dewarping as virtual cameras for debugging
+                        if show360DegAsVcs:
+                            vcBuffers.extend(fdBuffers)
+
+                        # Send the virtual camera images
+                        self.signalVirtualCameras.emit(vcBuffers, self.virtualCameraManager.getVirtualCameras())
+
+                frameTime = time.perf_counter() - currentTime
+                
+                if frameTime < frameTimeGoal:
+                    time.sleep(frameTimeGoal - frameTime)
 
                 prevTime = currentTime
+                currentTime = time.perf_counter()
+                actualFrameTime = currentTime - prevTime
+
+                frameTimeDelta += 1./self.fpsTarget - actualFrameTime
+
+                if np.abs(frameTimeDelta) < 1./(self.fpsTarget * 6):
+                    frameTimeGoal = 1./self.fpsTarget + frameTimeDelta
+                else:
+                    frameTimeGoal = 1./self.fpsTarget + 1./(self.fpsTarget * 6) * np.sign(frameTimeDelta)
```

Comment:
We should try to explain with a comment this equation, or put your numbers in variables or create a function with nice naming that explains computing done at lines 236 and 238, tt is difficult to understand.